### PR TITLE
[risk=low][RW-7089] Replace ProfileAccessModules with DataAccessPanel when DAR=true

### PR DESCRIPTION
**Pull request by jmthibault79**
Implementing https://projects.invisionapp.com/d/main#/console/21472057/455216387/preview

Clicking "Manage Data access" brings the user to the Data Access Requirements page, where equivalent functionality exists to complete modules.  This keeps the module completion code in one location.

If the CT is visible to users (e.g. in Test)

User has no access
no tiers

User has access to RT but not CT
RT-no-CT

User has access to both
yes-both

-----
If the CT is not visible to users (e.g. Prod)

User has RT access

yes-RT-only

User does not
no-RT

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/utils/access-tiers.tsx [hunk 1]
```diff
@@ -19,3 +21,8 @@ export enum AccessTierDisplayNames {
 export function hasRegisteredAccess(accessTierShortNames: Array<string>): boolean {
   return fp.includes(AccessTierShortNames.Registered, accessTierShortNames);
 }
+
+export const displayNameForTier = (shortName: string) => switchCase(shortName,
```
Comment (jmthibault79) [inline ui/src/app/utils/access-tiers.tsx:25]: anything besides one of these 2 will result in `undefined`

File: ui/src/app/pages/profile/data-access-panel.spec.tsx [hunk 2]
```diff
@@ -1,56 +1,67 @@
...
+  expect(findCTDenied(wrapper).exists()).toBeFalsy();
+}
 
 describe('Data Access Panel', () => {
-  const component = (props = {}) => {
+  const component = (props: DataAccessPanelProps) => {
     return mount(<DataAccessPanel {...props}/>);
   };
 
-  it('Should show controlled tier message while the user institution does not have an agreement', async() => {
```
Comment (jmthibault79) [inline ui/src/app/pages/profile/data-access-panel.spec.tsx:13]: These conditions no longer apply to the simplified DataAccessPanel.

File: ui/src/app/pages/profile/data-access-panel.tsx [hunk 3]
```diff
@@ -1,82 +1,72 @@
-import {StyledAnchorTag} from 'app/components/buttons';
-import {CheckCircle, ControlledTierBadge} from 'app/components/icons';
+import * as React from 'react';
```
Comment (jmthibault79) [inline ui/src/app/pages/profile/data-access-panel.tsx:1]: This new version does a good deal less than the previous, which is an unlaunched implementation of a previous version of the mocks.  See https://precisionmedicineinitiative.atlassian.net/browse/RW-6256 and https://github.com/all-of-us/workbench/pull/4803 for the history.

We'll probably add equivalent functionality back in for https://precisionmedicineinitiative.atlassian.net/browse/RW-7059 once we've finalized the appearance/behavior.

File: ui/src/app/pages/profile/profile-component.tsx [hunk 4]
```diff
@@ -111,9 +115,11 @@ export const ProfileComponent = fp.flow(
     async componentDidMount() {
       this.props.hideSpinner();
       try {
-        const details = await institutionApi().getPublicInstitutionDetails();
+        const publicInstitutionDetails = institutionApi().getPublicInstitutionDetails();
```
Comment (jmthibault79) [inline ui/src/app/pages/profile/profile-component.tsx:118]: These are Promises.  Do we have a particular naming convention for these?

File: ui/src/app/pages/profile/profile-component.tsx [hunk 5]
```diff
@@ -429,21 +435,23 @@ export const ProfileComponent = fp.flow(
             </div>
           </div>
           <div style={{width: '20rem', marginRight: '4rem'}}>
-            <div style={styles.title}>Free credits balance
+            <div style={{marginLeft: '1rem'}}>
```
Comment (jmthibault79) [inline ui/src/app/pages/profile/profile-component.tsx:438]: There was a visual alignment bug which became more obvious with this change.  It's probably easier to view this file with whitespace diffs off.

File: ui/src/app/pages/profile/profile-component.tsx [hunk 6]
```diff
@@ -429,21 +435,23 @@ export const ProfileComponent = fp.flow(
...
-                <FlexColumn style={{marginLeft: '0.8rem'}}>
-                    <div style={{marginTop: '0.4rem'}}><AoU/> free credits used:</div>
-                    <div>Remaining <AoU/> free credits:</div>
-                </FlexColumn>
-                <FlexColumn style={{alignItems: 'flex-end', marginLeft: '1.0rem'}}>
-                  <div style={{marginTop: '0.4rem', fontWeight: 600}}>{formatFreeCreditsUSD(profile.freeTierUsage)}</div>
-                  <div style={{fontWeight: 600}}>{formatFreeCreditsUSD(profile.freeTierDollarQuota - profile.freeTierUsage)}</div>
-                </FlexColumn>
-            </FlexRow>}
-            {/* controlledTierEnabled && <DataAccessPanel tiers={profile.accessTierShortNames}/> */}
```
Comment (jmthibault79) [inline ui/src/app/pages/profile/profile-component.tsx:445]: the original DataAccessPanel was never launched

**Comment by jmthibault79** [comment]
Waiting on PO+UX approval

**Comment by jmthibault79** [comment]
e2e failure is a notebooks test

File: ui/src/app/pages/profile/data-access-panel.tsx (hunk 3 above)
Comment (als364) [reply ui/src/app/pages/profile/data-access-panel.tsx:1]: I do not have access to the new mocks but I believe you that this matches them 🤷

File: ui/src/app/pages/profile/data-access-panel.tsx [hunk 7]
```diff
@@ -1,82 +1,67 @@
...
-    tiers = [], hasInstitutionalAgreement = false, userRevoked = false
-  }) => {
-  const sectionId = Utils.useId();
-  return <section aria-labelledby={sectionId}>
-    <div id={sectionId} style={styles.title}>Data access</div>
+  const sectionId = useId();
+  return <section aria-labelledby={sectionId} style={{marginLeft: '1rem'}}>
+    <FlexRow id={sectionId}>
+      <div style={styles.title}>Data access</div>
+      <Link style={{marginLeft: 'auto'}} onClick={() => navigate(['data-access-requirements'])}>Manage data access</Link>
```
Comment (als364) [inline ui/src/app/pages/profile/data-access-panel.tsx:58]: Please use the react-router [Link](https://reactrouter.com/web/api/Link) for navigation instead.

Comment (jmthibault79) [reply ui/src/app/pages/profile/data-access-panel.tsx:58]: ok, thanks.

Do we always want this where `navigate` is currently used?

Comment (als364) [reply ui/src/app/pages/profile/data-access-panel.tsx:58]: Yes, we do. There's a low priority [ticket](https://precisionmedicineinitiative.atlassian.net/browse/RW-7168) to replace them all.

**Review by als364 — APPROVED** [review]

File: ui/src/app/pages/profile/data-access-panel.tsx (hunk 7 above)
Comment (jmthibault79) [reply ui/src/app/pages/profile/data-access-panel.tsx]: updated here

**Pull request merged**